src/node/interfaces.cpp

```diff
@@ -866,7 +868,22 @@ class BlockTemplateImpl : public BlockTemplate
                                                     m_block_template(std::move(block_template)),
                                                     m_node(node)
     {
-        assert(m_block_template);
+        // Don't track the dummy coinbase, because it can be modified in-place
+        // by submitSolution()
+        for (const CTransactionRef& tx : Assert(m_block_template)->block.vtx | std::views::drop(1)) {
+            m_node.template_tx_refs[tx]++;
+        }
+    }
+
+    ~BlockTemplateImpl()
+    {
+        for (const CTransactionRef& tx : m_block_template->block.vtx | std::views::drop(1)) {
+            auto ref_count{m_node.template_tx_refs.find(tx)};
+            if (!Assume(ref_count != m_node.template_tx_refs.end())) break;
+            if (--ref_count->second == 0) {
```

Comment:
Just ran a mutation testing for this PR and this is the only unkilled mutant - feel free to ignore if it doesn't make sense to address:

```diff
diff --git a/src/node/interfaces.cpp b/src/node/interfaces.cpp
index f734296b24..13ddbb672f 100644
--- a/src/node/interfaces.cpp
+++ b/src/node/interfaces.cpp
@@ -880,7 +880,7 @@ public:
         for (const CTransactionRef& tx : m_block_template->block.vtx | std::views::drop(1)) {
             auto ref_count{m_node.template_tx_refs.find(tx)};
             if (!Assume(ref_count != m_node.template_tx_refs.end())) break;
-            if (--ref_count->second == 0) {
+            if (1==1) {
                 m_node.template_tx_refs.erase(ref_count);
             }
         }
```

Reply:
Nice, thank you!